### ISSUE TITLE
sudoers: `parse_one_default_setting` strips ONE `!`, but sudo TOGGLES per sigil, so `Defaults !!!authenticate` silently loses a STIG weakening

### ISSUE THREAD

**Issue by ErstBlack** [rulesteward/rulesteward#697]
`parse_one_default_setting` (`crates/rulesteward-sudoers/src/parser.rs:502`) reads the negation sigil with a single `strip_prefix('!')`:

```rust
let (negated, body) = match token.strip_prefix('!') {
    Some(rest) => (true, rest.trim_start()),
    None => (false, token),
};
```

Real sudo does not strip one - it **toggles per sigil**, so PARITY decides. Grounded on this host 2026-08-19, sudo 1.9.17p2, all three `visudo -c -f -` **rc 0**:

| input | `cvtsudoers -f json` |
|---|---|
| `Defaults !authenticate` | `{"authenticate": false}` |
| `Defaults !!authenticate` | `{"authenticate": **true**}` |
| `Defaults !!!authenticate` | `{"authenticate": false}` |

This is the same "not-not-X is X" rule the principal positions follow (`(!!root)` reports plain `{"username":"root"}` with no `negated` flag). The difference is that in `Defaults` position the resulting SENSE is load-bearing for a lint, and here it is lost.

## The defect

| input | sudo's truth | RuleSteward |
|---|---|---|
| `Defaults !authenticate` | authenticate=false, a STIG weakening | 4x `sudo-W04` - the weakening IS reported (correct) |
| `Defaults !!authenticate` | authenticate=true, NOT a weakening | 3x `sudo-W04` - correctly silent |
| `Defaults !!!authenticate` | authenticate=false, **a STIG weakening** | 3x `sudo-W04` - **the weakening is MISSED** |

The third row is a fail-open. `sudo-W04` is the sudo hardening baseline check (`!authenticate` is one of its named weakenings), so an odd run of three or more sigils hides a real weakening on a file `visudo` accepts.

Mechanism: `strip_prefix` removes one `!` and sets `negated = true`, leaving `body = "!!authenticate"`. The name then never matches `authenticate`, so the baseline comparison does not fire. The middle row is correct only by accident - the name fails to match there too, and not-matching happens to be the right answer.

## The one-byte control

The rows above ARE the control set: they differ by a single `!` each and ground truth alternates, while RuleSteward's output does not. Row 1 vs row 3 is the sharpest pair - identical sense (both `false`, both weakenings), opposite RuleSteward answers.

## Where this came from

Found by the fix-then-sweep for the negation-sigil concept while closing #670 / #671 / #672. Those three covered the four PRINCIPAL-position call sites; this is a fifth call site in `Defaults` position that the sweep surfaced:

```
$ git grep -nE "trim_start_matches\('!'\)|strip_prefix\('!'\)" -- crates/rulesteward-sudoers/src
lints/aliases.rs:78,159,283,361,384,478   trim_start_matches   (correct)
lints/tokens/command_specs.rs:70          trim_start_matches   (correct)
lints/tokens/defaults.rs:337              trim_start_matches   (correct)
lints/tokens/runas.rs:38                  trim_start_matches   (fixed by #671)
parser.rs:502                             strip_prefix         <- THIS
```

`aliases.rs:69` already carries a comment saying a single `strip_prefix('!')` is wrong and mis-classifies - that reasoning applies here too, and `parser.rs:502` is the last site that still does it.

## Fix direction

Count the leading sigil run and use its PARITY for `negated`, trimming the whole run from `body` - not `trim_start_matches` alone, because unlike the validity scans in the other call sites this one needs the sense, not just a clean token.

Any fix needs all three rows above as tests, since a trim-all implementation gets rows 1 and 3 right and row 2 wrong, and the current strip-one gets rows 1 and 2 right and row 3 wrong. Neither is caught by a single row.
